src/vmm/src/vmm_config/machine_config.rs

```diff
@@ -217,6 +231,10 @@ impl VmConfig {
             Some(other) => Some(CpuTemplateType::Static(other)),
         };
 
+        if page_config.is_hugetlbfs() && KernelVersion::get()? < KernelVersion::new(5, 10, 0) {
```

Comment:
is there a reason we cut 4.16-5.9 kernel users from the feature?

Reply:
Generally, I wanted to avoid avoid mentioning kernel versions that Firecracker does not support anywhere. Can change it to 
```suggestion
        if page_config.is_hugetlbfs() && KernelVersion::get()? < KernelVersion::new(4, 16, 0) {
```
if people feel strongly about it

Reply:
I'm being biased by https://github.com/firecracker-microvm/firecracker/blob/826831d566f54fa88b3fe6d672a0d5f79ba35258/src/utils/src/kernel_version.rs#L89 . If we followed the same logic, we would only support io_uring on 6.1 onwards. Otherwise no strong opinion either.

Reply:
Mh, I guess that's slightly different though, because 5.10 is a supported version, so there we're making sure that Firecracker does not try to use io_uring on a patch version where support for it isnt present

Reply:
Changed the rust code to explicitly refer to 4.16